packages/drivers/routerlicious-driver/src/errorUtils.ts

```diff
@@ -76,43 +77,56 @@ export function createR11sNetworkError(
 			// the error message will start with NetworkError as defined in restWrapper.ts
 			// If there exists a self-signed SSL certificates error, throw a NonRetryableError
 			// TODO: instead of relying on string matching, filter error based on the error code like we do for websocket connections
-			if (errorMessage.includes("failed, reason: self signed certificate")) {
-				return new NonRetryableError(
-					errorMessage,
-					RouterliciousErrorType.sslCertError,
-					props,
-				);
-			}
-			return new GenericNetworkError(
-				errorMessage,
-				errorMessage.startsWith("NetworkError"),
-				props,
-			);
+			error = errorMessage.includes("failed, reason: self signed certificate")
+				? new NonRetryableError(errorMessage, RouterliciousErrorType.sslCertError, props)
+				: new GenericNetworkError(
+						errorMessage,
+						errorMessage.startsWith("NetworkError"),
+						props,
+				  );
+			break;
 		case 401:
 		// The first 401 is manually retried in RouterliciousRestWrapper with a refreshed token,
 		// so we treat repeat 401s the same as 403.
 		case 403:
-			return new AuthorizationError(errorMessage, undefined, undefined, props);
+			error = new AuthorizationError(errorMessage, undefined, undefined, props);
+			break;
 		case 404:
 			const errorType = RouterliciousErrorType.fileNotFoundOrAccessDeniedError;
-			return new NonRetryableError(errorMessage, errorType, props);
+			error = new NonRetryableError(errorMessage, errorType, props);
+			break;
 		case 429:
-			return createGenericNetworkError(errorMessage, { canRetry: true, retryAfterMs }, props);
+			error = createGenericNetworkError(
+				errorMessage,
+				{ canRetry: true, retryAfterMs },
+				props,
+			);
+			break;
 		case 500:
 		case 502:
-			return new GenericNetworkError(errorMessage, true, props);
+			error = new GenericNetworkError(errorMessage, true, props);
+			break;
 		default:
 			const retryInfo = { canRetry: retryAfterMs !== undefined, retryAfterMs };
-			return createGenericNetworkError(errorMessage, retryInfo, props);
+			error = createGenericNetworkError(errorMessage, retryInfo, props);
+			break;
 	}
+	error.endpointReachable = endpointReachable;
```

Comment:
I have not traced code, but similar question / feedback - should it be set to true here unconditionally?

Reply:
This is not true for r11s error. The function also handles/produces error in case status code is undefined or not supplied i.e endpoint was not reachable. So we can not make that change here.